Server/src/controllers/getDatasetController.ts

```diff
@@ -22,17 +22,17 @@ export class getDataController {
     async createRequestForData(request: Request, response: Response) {
         let validateData = this.validateInputData(request)
         if (!validateData) {
-            response.status(400).send("Invalid search params entered");
+            response.status(400).json("Invalid search params entered");
         }
         else {
             let requestParams: any = { ...request.query };
             this.processedRequest = requestParams;
             try {
                 const retrieveDataObject = new retrieveData();
                 let arrayOfData = await retrieveDataObject.getArrayOfDatasets(this.processedRequest)
-                return response.status(200).send(arrayOfData);
+                return response.status(200).json(arrayOfData);
             } catch (err) {
-                response.status(400).send(err);
+                response.status(500).json(err);

```

Comment:
Should be err.status and err.message if we are using the BadRequest and InternalServer errors weve imported